cmd/node/factory/structsStatusHandlers.go

```diff
@@ -63,19 +73,28 @@ func CreateStatusHandlers(arguments *ArgStatusHandlers) (*statusHandlersInfo, er
 
 	useTermui := !arguments.Ctx.GlobalBool(arguments.LogViewName)
 	if useTermui {
-		views, err = createViews(presenterStatusHandler)
+		views, err = createViews(presenterStatusHandler, arguments.ChanStartViews)
 		if err != nil {
 			return nil, err
 		}
 
-		writer, ok := presenterStatusHandler.(io.Writer)
-		if ok {
-			logger.ClearLogObservers()
-			err = logger.AddLogObserver(writer, &logger.PlainFormatter{})
-			if err != nil {
-				return nil, err
+		go func() {
+			<-arguments.ChanLogRewrite
+			writer, ok := presenterStatusHandler.(io.Writer)
+			if ok {
+				logger.ClearLogObservers()
+				err = logger.AddLogObserver(writer, &logger.PlainFormatter{})
+				if err != nil {
+					log.Warn("cannot add log observer for TermUI", "error", err)
+				}
+				if arguments.LogFile != nil {
```

Comment:
change the condition and do an early return

Reply:
done